force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -54,50 +60,78 @@ public with sharing class FlowControlLoopsCollections {
      * @param num2 The second number.
      * @return The maximum of the two numbers, or null if either number is null.
      */
-    public static Integer findMax(Integer num1, Integer num2) {
-        // Initialize largestNum as null
-        Integer largestNum = null;
-
-        return null; // Replace null with the variable you used to store the result
-    }
-
+        public static Integer findMax(Integer num1, Integer num2) {
+            //Check if either num1 or num2 is null
+            if (num1 == null || num2 == null){
+                return null;
+
+            }
+            Integer largestNum;
+            if (num1 > num2){
+                largestNum = num1;
+            } else {
+                largestNum = num2;
+            }
+            return largestNum;
+        }
 
-    /**
+    /** Q4
      * Question 4
      * Checks if a number is positive, negative, or zero.
      * If the number is null, return null.
      * Example: checkNumber(5) should return "Positive"
      * @param a The number to check.
      * @return A string indicating whether the number is "Positive", "Negative", or "Zero", or null if the number is null.
      */
-    public static String checkNumber(Integer a) {
-        return null; // Replace null with the variable you used to store the result
-    }
+        public static String checkNumber(Integer a) {
+        if (a == null) {
+            return null;
+        }
+        if (a > 0) {
+            return 'Positive';
+        } else if (a < 0) {
+            return 'Negative';
+        } else {
+            return 'Zero';
+        }
+        }
```

Comment:
Watch your identation! Always indent after open curly braces, and then bring it back after a closed curly brace:
```suggestion
        public static String checkNumber(Integer a) {
             if (a == null) {
                 return null;
             }
             if (a > 0) {
                 return 'Positive';
             } else if (a < 0) {
                 return 'Negative';
             } else {
                  return 'Zero';
        }
        }
```